app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -281,6 +290,34 @@ class DefaultSessionControlController(
         )
     }
 
+    override fun handleRenameTopSiteClicked(topSite: TopSite) {
+        if (topSite.id != null) {
```

Comment:
You're right. I added a check because I was not 100% sure that `isPinned = topSite.type != FRECENT` ensures we have a database entry, but it seems obvious now.